### ISSUE TITLE
[@gentype.import] is not working correctly with React.FC

### ISSUE THREAD

**Issue by lalnuo** [cristianoc/genType#207]
Latest gentype upgrade that bundles the react bindings is breaking my build. 

Typescript component:

```typescript
import React from 'react';

type Props = {
  foo: string;
};

const Foo: React.FC<Props> = ({ foo }) => <div>{foo}</div>;

export default Foo;
```

Binding: 
```
module Foo = {
  [@genType.import ("./Foo", "default")]
  external make: (~foo: string) => React.element = "";
};
```

Gentype output:
```typescript
/* TypeScript file generated by genType. */
/* eslint-disable import/first */


import {default as makeNotChecked} from './Foo';

// In case of type error, check the type of 'make' in 'Foo.re' and './Foo'.
export const makeTypeChecked: (_1:{ readonly foo: string }) => JSX.Element = makeNotChecked;

// Export 'make' early to allow circular import from the '.bs.js' file.
export const make: unknown = function (Argfoo: any) {
  const result = makeTypeChecked({foo:Argfoo});
  return result
} as (_1:{ readonly foo: string }) => JSX.Element;
```

Typescript error:
```
ERROR in /Users/lallinuo/projects/foo/bindings/Foo.gen.tsx(8,14):
TS2322: Type 'FunctionComponent<Props>' is not assignable to type '(_1: { readonly foo: string; }) => Element'.
  Type 'ReactElement<any, string | ((props: any) => ReactElement<any, string | ... | (new (props: any) => Component<any, any, any>)> | null) | (new (props: any) => Component<any, any, any>)> | null' is not assignable to type 'Element'.
    Type 'null' is not assignable to type 'Element'.
```

**Comment by cristianoc** [comment]
@lalnuo what was your previous output? Could be that the function component was not recognized beforehand.

(Also, I guess the binding is missing a `[@react.component]`).

In general it seems that a function type can be converted to `React.FC<...>`, but conversion the other way is not possible.
So it could be that the more permissive typing should be generated instead: would this have worked for you?
```ts
export const makeTypeChecked: React.FC<{ readonly foo: string }> = makeNotChecked;
```

**Comment by lalnuo** [comment]
@cristianoc 
```
export const makeTypeChecked: React.FC<{ readonly foo: string }> = makeNotChecked;
```

would work for me. That makes the TS error go away.

**Comment by cristianoc** [comment]
@lalnuo great -- I'll do some checks to see if the change would affect anything else, and confirm the impression that it should enable strictly more type checks to pass. If all is OK, I'll do the change.

**Comment by lalnuo** [comment]
Thank you so much @cristianoc 👍

**Comment by cristianoc** [comment]
This is tracked in https://github.com/cristianoc/genType/pull/208.

**Comment by cristianoc** [comment]
Addressed in release 2.29.1.
